swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/BillableUsageConsumer.java

```diff
@@ -201,8 +238,12 @@ public AzureUsageContext lookupAzureUsageContext(BillableUsage billableUsage)
           log.warn("Subscription could not be determined due to ambiguois billingAccountId");
```

Comment:
This is actually the place where we want to throw the error/resend the billableUsage. This condition will catch only the exceptions where the azure subscription can not be determined and is missing part of the billingAccountId.

Reply:
I think these got a little messed up between services. 

Subscription Service has 
```
SUBSCRIPTION_RECENTLY_TERMINATED(
      1005, "Subscription recently terminated. No active subscriptions."),
 SUBSCRIPTION_CANNOT_BE_DETERMINED(
      1006,
      "Multiple possible matches found. Likely a subscription is missing part of its billingAccountId. (check azureSubscriptionId)")
```
https://github.com/RedHatInsights/rhsm-subscriptions/blob/3783b5f57f571fce90af4dc8429b4d638cc464fa/swatch-core/src/main/java/org/candlepin/subscriptions/exception/ErrorCode.java#L53

And swatch-produce-azure has:
```
AZURE_MANUAL_SUBMISSION_DISABLED(1005, "Manual submission disabled."),
  SUBSCRIPTION_RECENTLY_TERMINATED(1006, "Subscription recently terminated"),
```
https://github.com/RedHatInsights/rhsm-subscriptions/blob/3783b5f57f571fce90af4dc8429b4d638cc464fa/swatch-producer-azure/src/main/java/com/redhat/swatch/azure/exception/ErrorCode.java#L30

I'm thinking we should set SUBSCRIPTION_RECENTLY_TERMINATED to 1005 on swatch-azure-producer and move AZURE_MANUAL_SUBMISSION_DISABLED to a different error code(2001 maybe?)

Reply:
Hmm I think we should stick with just 1006 for now and we can flesh out other failures further in the future.

Reply:
@kflahert PR updated with this change.
In the end, I didn't change the error code, because it was my mistake to match the 1006 in the azure error codes. So, for now, let's keep using as it was (but yes, ideally, the same codes should be shared across all the services).